client/signup/controller.js

```diff
@@ -331,6 +331,10 @@ export default {
 			await loadExperimentAssignment( 'calypso_signup_monthly_plans_default_202201_v2' );
 		}
 
+		if ( 'onboarding' === flowName ) {
```

Comment:
Shouldn't we use the same condition for prefetching as we use for evaluating the assignment?
```suggestion
		if ( isMobile() && 'onboarding' === flowName ) {
```